### PR TITLE
Reimplement MakePriority

### PR DESCRIPTION
**Pull request by spencerkimball**
This is more sensical than the previous approach and allows smaller
priorities to be set when you want to, for example, de-prioritize
transactions, such as for batch processing jobs.

Fixes issue #3643.

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/3659)

### CODE REVIEW

File: roachpb/data_test.go [hunk 1]
```diff
@@ -457,20 +475,21 @@ func TestTransactionUpdate(t *testing.T) {
 
 }
 
-func TestIsPrev(t *testing.T) {
-	for i, tc := range []struct {
-		k, m Key
-		ok   bool
+// TestMakePriority verifies that setting user priority of P results
+// in MakePriority returning priorities that are P times more likely
```
Comment (tamird) [inline roachpb/data_test.go]: this comment appears to refer to an old version of this test

File: storage/replica_test.go [hunk 2]
```diff
@@ -223,8 +223,7 @@ func newTransaction(name string, baseKey roachpb.Key, userPriority int32,
 		offset = clock.MaxOffset().Nanoseconds()
 		now = clock.Now()
 	}
-	return roachpb.NewTransaction(name, baseKey, userPriority,
-		isolation, now, offset)
+	return roachpb.NewTransaction(name, baseKey, userPriority, isolation, now, offset)
```
Comment (tamird) [inline storage/replica_test.go]: seems unrelated

File: roachpb/data_test.go [hunk 3]
```diff
@@ -158,6 +158,24 @@ func TestNextKey(t *testing.T) {
 	}
 }
 
+func TestIsPrev(t *testing.T) {
```
Comment (tamird) [inline roachpb/data_test.go]: this diff was confusing, looks like this test moved for no reason?

File: roachpb/api.proto [hunk 4]
```diff
@@ -584,16 +584,15 @@ message Header {
...
-  // Commands with user_priority=100 will be 100x less likely to be
-  // aborted as conflicting transactions or non-transactional commands
-  // with user_priority=1. This value is ignored if Txn is
+  // user_priority allows an explicit priority to be specified for
+  // commands. This value should be a positive integer [1, 2^31-1].
+  // If set to 0 or not set, a random value in the interval is
+  // chosen. To set a lower than average priority, use the interval
+  // [1, 2^30). To set a higher than average priority, use the
+  // interval [2^30, 2^31-1]. This value is ignored if Txn is
   // specified. If neither this value nor txn is specified, the value
```
Comment (tamird) [inline roachpb/api.proto]: s/txn/Txn/ for consistency with the preceding sentence

File: roachpb/api.proto [hunk 5]
```diff
@@ -584,16 +584,15 @@ message Header {
...
       (gogoproto.customname) = "RangeID", (gogoproto.casttype) = "RangeID"];
-  // user_priority specifies priority multiple for non-transactional
-  // commands. This value should be a positive integer [1, 2^31-1).
-  // It's properly viewed as a multiple for how likely this
-  // transaction will be to prevail if a write conflict occurs.
-  // Commands with user_priority=100 will be 100x less likely to be
-  // aborted as conflicting transactions or non-transactional commands
-  // with user_priority=1. This value is ignored if Txn is
+  // user_priority allows an explicit priority to be specified for
+  // commands. This value should be a positive integer [1, 2^31-1].
```
Comment (tamird) [inline roachpb/api.proto]: `[1, 2^31)` here and on L592 for consistency with the notation used on L591

**Comment by tamird** [comment]
LGTM

File: roachpb/api.proto [hunk 6]
```diff
@@ -584,16 +584,15 @@ message Header {
...
+  // If set to 0 or not set, a random value in the interval is
+  // chosen. To set a lower than average priority, use the interval
+  // [1, 2^30). To set a higher than average priority, use the
+  // interval [2^30, 2^31-1]. This value is ignored if Txn is
   // specified. If neither this value nor txn is specified, the value
-  // defaults to 1.
-  optional int32 user_priority = 4 [default = 1];
+  // defaults to 0, and a random priority is chosen.
+  optional int32 user_priority = 4;
```
Comment (bdarnell) [inline roachpb/api.proto]: Now that zero is equivalent to not-set, this field can be made non-nullable, which should simplify its usage.

Comment (tamird) [inline roachpb/api.proto]: Now that this field doesn't have a default, this message's getters can be removed https://github.com/cockroachdb/cockroach/blob/spencerkimball/fix-max-priority/roachpb/api.proto#L573:L575

File: roachpb/data.go [hunk 7]
```diff
@@ -602,32 +602,21 @@ func (t *Transaction) IsInitialized() bool {
...
 func MakePriority(userPriority int32) int32 {
-	// A currently undocumented feature allows an explicit priority to
-	// be set by specifying priority < 1. The explicit priority is
-	// simply -userPriority in this case. This is hacky, but currently
-	// used for unittesting. Perhaps this should be documented and allowed.
+	// A former undocumented feature allowed an explicit priority to be
+	// set by specifying priority < 1. The explicit priority is simply
+	// -userPriority in this case. This is used for unittesting, before
+	// the specified priority was taken at face value, so we maintain
+	// the old behavior.
```
Comment (bdarnell) [inline roachpb/data.go]: There aren't that many tests that use negative priorities, are there? I think it would be nice to clean those up and get rid of this special case.

File: roachpb/api.proto (hunk 4 above)
Comment (tamird) [inline roachpb/api.proto]: "the value defaults to 0" doesn't make much sense now

**Comment by bdarnell** [comment]
LGTM

**Comment by tschottdorf** [comment]
LGTM after others' comments

---

Reviewed 9 of 9 files at r1.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[roachpb/data_test.go, line 479 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7OKcV2-T5FZXVKs1bA-r1-479:-K7RVLk7mABSuPyRH5YD:1203882245) ([raw file](https://github.com/cockroachdb/cockroach/blob/c4b7420396de5550bd0374ce55faecd2cdfca624/roachpb/data_test.go#L479)):*
yeah, and while the existing test cases go away with the cleanup, it doesn't check the zero user priority.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-:-K7RVfJTFACId-kPy9CM:-703740037)*

**Comment by spencerkimball** [comment]
I changed my mind on this. I've updated the associated PR to change user priority to a floating point value, which now allows us to bias to lower operation priorities (e.g. 0.1x). The floating point value is easier to reason about than just having an explicit int32 and it preserves the randomness between different operations with the same priorities.

File: roachpb/data_test.go (hunk 3 above)
Comment (spencerkimball) [inline roachpb/data_test.go]: IsPrev() is a test for keys and it was out of place in the test file. Just a little housekeeping.

**Comment by tamird** [comment]
Reviewed 9 of 9 files at r1, 14 of 14 files at r2.
Review status: all files reviewed at latest revision, 15 unresolved discussions, some commit checks failed.

---

*[client/client_test.go, line 148 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7RosfwEEPXYJi4lecJ:-K7RosfwEEPXYJi4lecK:-1533642829) ([raw file](https://github.com/cockroachdb/cockroach/blob/84000e759288e4affd19ee84a602777e66979f95/client/client_test.go#L148)):*
float here, rather than cast below?

---

*[client/db.go, line 39 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7Rp-yKSySDNgRAbMJb:-K7Rp-yKSySDNgRAbMJc:-1934042506) ([raw file](https://github.com/cockroachdb/cockroach/blob/84000e759288e4affd19ee84a602777e66979f95/client/db.go#L39)):*
i think you can unexport this

---

*[roachpb/api.proto, line 593 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7RpS8WP4n51T-UoRIk:-K7RpS8WP4n51T-UoRIl:1024918667) ([raw file](https://github.com/cockroachdb/cockroach/blob/84000e759288e4affd19ee84a602777e66979f95/roachpb/api.proto#L593)):*
s/txn/Txn/

---

*[roachpb/api.proto, line 594 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7RpdS9UN_sum-kqg3v:-K7RpdS9UN_sum-kqg3w:-2146073204) ([raw file](https://github.com/cockroachdb/cockroach/blob/84000e759288e4affd19ee84a602777e66979f95/roachpb/api.proto#L594)):*
s/, and a random priority is chosen//

---

*[roachpb/api.proto, line 597 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7RpvmtKMUY3KBleslt:-K7RpvmtKMUY3KBleslu:-548744827) ([raw file](https://github.com/cockroachdb/cockroach/blob/84000e759288e4affd19ee84a602777e66979f95/roachpb/api.proto#L597)):*
does this need to be `double` rather than `float?

---

*[roachpb/data.go, line 609 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7Rq8J0ym3LLt7GQTm8:-K7Rq8J0ym3LLt7GQTm9:-1379048617) ([raw file](https://github.com/cockroachdb/cockroach/blob/84000e759288e4affd19ee84a602777e66979f95/roachpb/data.go#L609)):*
what is "probabilistically greater"?

---

*[roachpb/data.go, line 629 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7RqRIskqWISSP5CXZq:-K7RqRIskqWISSP5CXZr:-307353406) ([raw file](https://github.com/cockroachdb/cockroach/blob/84000e759288e4affd19ee84a602777e66979f95/roachpb/data.go#L629)):*
shouldn't these ranges be `[1, 3^31)`?

---

*[roachpb/data.go, line 693 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7RqjCf19PvDi4i0w_q:-K7RqjCf19PvDi4i0w_r:166874) ([raw file](https://github.com/cockroachdb/cockroach/blob/84000e759288e4affd19ee84a602777e66979f95/roachpb/data.go#L693)):*
?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659)*

**Comment by petermattis** [comment]
Review status: all files reviewed at latest revision, 10 unresolved discussions, some commit checks failed.

---

*[roachpb/data.go, line 613 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7OKcV2-T5FZXVKs1b9-r1-613:-K7RyjRqluo6ie_FTh3C:-729945980) ([raw file](https://github.com/cockroachdb/cockroach/blob/c4b7420396de5550bd0374ce55faecd2cdfca624/roachpb/data.go#L613)):*
I think the tests that do use negative priorities want to avoid the randomization. 

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659)*

**Comment by spencerkimball** [comment]
Review status: all files reviewed at latest revision, 10 unresolved discussions, some commit checks failed.

---

*[roachpb/api.proto, line 588 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7OKcV2-T5FZXVKs1b8-r1-588:-K7RyTtCPMsmYFwUstKM:-2132980318) ([raw file](https://github.com/cockroachdb/cockroach/blob/c4b7420396de5550bd0374ce55faecd2cdfca624/roachpb/api.proto#L588)):*
This is out of date.

---

*[roachpb/api.proto, line 593 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7OKcV2-T5FZXVKs1b8-r1-593:-K7RyhXiLOAb9G4eoqxA:1552595983) ([raw file](https://github.com/cockroachdb/cockroach/blob/c4b7420396de5550bd0374ce55faecd2cdfca624/roachpb/api.proto#L593)):*
Comment out of date.

---

*[roachpb/api.proto, line 595 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7OKcV2-T5FZXVKs1b8-r1-595:-K7Rz2LnolVMjwtLd1Y4:-392814165) ([raw file](https://github.com/cockroachdb/cockroach/blob/c4b7420396de5550bd0374ce55faecd2cdfca624/roachpb/api.proto#L595)):*
These comments are out of date. We still do need the default value of 1.

---

*[roachpb/api.proto, line 597 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7RpvmtKMUY3KBleslt:-K7Ryq6EwyFSeE6P5E2N:862532833) ([raw file](https://github.com/cockroachdb/cockroach/blob/84000e759288e4affd19ee84a602777e66979f95/roachpb/api.proto#L597)):*
No, but I prefer using 64 bit for floating point unless there's some sort of size constraint.

---

*[roachpb/data.go, line 613 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7OKcV2-T5FZXVKs1b9-r1-613:-K7Rz8rkQqe7JOz_EJfw:-1995676254) ([raw file](https://github.com/cockroachdb/cockroach/blob/c4b7420396de5550bd0374ce55faecd2cdfca624/roachpb/data.go#L613)):*
With the updated change, we do need the negative priorities still.

---

*[roachpb/data.go, line 629 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7RqRIskqWISSP5CXZq:-K7RzMe0wz5TLGQ5PFsf:-1905315078) ([raw file](https://github.com/cockroachdb/cockroach/blob/84000e759288e4affd19ee84a602777e66979f95/roachpb/data.go#L629)):*
What do you mean 3^31?

---

*[roachpb/data.go, line 693 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7RqjCf19PvDi4i0w_q:-K7Rzi9qApIvwBhoz_Vs:-420569041) ([raw file](https://github.com/cockroachdb/cockroach/blob/84000e759288e4affd19ee84a602777e66979f95/roachpb/data.go#L693)):*
I tried to clarify the comment.

---

*[roachpb/data_test.go, line 479 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7OKcV2-T5FZXVKs1bA-r1-479:-K7RzrPA6oHpFY0A0lqp:-251652152) ([raw file](https://github.com/cockroachdb/cockroach/blob/c4b7420396de5550bd0374ce55faecd2cdfca624/roachpb/data_test.go#L479)):*
The latest change doesn't have these issues.

---

*[storage/replica_test.go, line 226 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7OKcV4j6dhsY_8FtDN-r1-226:-K7S--4Q1MRSSkaroZoQ:1291823442) ([raw file](https://github.com/cockroachdb/cockroach/blob/c4b7420396de5550bd0374ce55faecd2cdfca624/storage/replica_test.go#L226)):*
It's unrelated. I just dislike 80 character line breaks in Go.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659)*

**Comment by tschottdorf** [comment]
LGTM. I also like the 10x concept.

---

Reviewed 1 of 14 files at r2.
Review status: 8 of 14 files reviewed at latest revision, 17 unresolved discussions, some commit checks failed.

---

*[roachpb/api.proto, line 597 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7S55L9AA43uuMiMUwf:-K7S55L9AA43uuMiMUwg:962824995) ([raw file](https://github.com/cockroachdb/cockroach/blob/84000e759288e4affd19ee84a602777e66979f95/roachpb/api.proto#L597)):*
we shouldn't use default values, they have been dropped in proto3 (to the best of my knowledge).

---

*[roachpb/data.go, line 609 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7SOoBRNVE-qvGXgScI:-K7SOoBpnfsPc2Ptnu36:1069930778) ([raw file](https://github.com/cockroachdb/cockroach/blob/84000e759288e4affd19ee84a602777e66979f95/roachpb/data.go#L609)):*
what does the word "probabilistically" add here?

---

*[roachpb/data.go, line 612 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7SQE8O1t1uDrKCffB-:-K7SQE8O1t1uDrKCffB0:-1160524697) ([raw file](https://github.com/cockroachdb/cockroach/blob/84000e759288e4affd19ee84a602777e66979f95/roachpb/data.go#L612)):*
This doesn't sound right. The calculation below seems to say `userPriority = probability of beating a random int32 / probability of losing against a random int32`.
So instead of comparing to userPriority 1:

> If userPriority=100, the likelihood of winning against a random number is 100 times that of losing against one. If userPriority=.01, losses will be 100x more common than wins. Balance is achieved when userPriority=1, in which case the priority chosen is unbiased.

---

*[roachpb/data.go, line 641 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7S6GcbCZum5zfWaIAL:-K7S6GcbCZum5zfWaIAM:1103911742) ([raw file](https://github.com/cockroachdb/cockroach/blob/84000e759288e4affd19ee84a602777e66979f95/roachpb/data.go#L641)):*
`p` is an awkward name for `user priority` since it's the standard variable for a probability.

---

*[roachpb/data.go, line 693 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7RqjCf19PvDi4i0w_q:-K7SVoL7MZ9VT6W9EmnL:1019550419) ([raw file](https://github.com/cockroachdb/cockroach/blob/84000e759288e4affd19ee84a602777e66979f95/roachpb/data.go#L693)):*
wasn't the old one clearer though?

---

*[roachpb/data_test.go, line 508 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7SWAEGJx1MdVaZCIdY:-K7SWAEHRHgyKsEPRWjz:-567421895) ([raw file](https://github.com/cockroachdb/cockroach/blob/84000e759288e4affd19ee84a602777e66979f95/roachpb/data_test.go#L508)):*
remove this.

---

*[roachpb/data_test.go, line 509 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7SXGrJ7rHX65WJ3wFF:-K7SXGrJ7rHX65WJ3wFG:-951227918) ([raw file](https://github.com/cockroachdb/cockroach/blob/84000e759288e4affd19ee84a602777e66979f95/roachpb/data_test.go#L509)):*
`if m := diff/float64(userPri); m > 1 {` and then re-use `m` below.

---

*[roachpb/data_test.go, line 523 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7SXzhJVXna2QfrPK4K:-K7SXzhKtR6SL8NWCBxs:46671867) ([raw file](https://github.com/cockroachdb/cockroach/blob/84000e759288e4affd19ee84a602777e66979f95/roachpb/data_test.go#L523)):*
can you sneak `0 -> 0` in here so that it's exercised anywhere? Regarding my earlier comment about defaults, it could be acceptable to map `0 -> 1` in `MakePriority` so that "unspecified" goes to "random" since nobody needs to use `0` anyway.

---

*[roachpb/data_test.go, line 564 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7SYpxBQm72KxPvlfcX:-K7SYpxBQm72KxPvlfcY:2133622994) ([raw file](https://github.com/cockroachdb/cockroach/blob/84000e759288e4affd19ee84a602777e66979f95/roachpb/data_test.go#L564)):*
this test and the above seem likely they would combine nicely (two maps, and split at `1`). There's also copy-pasta `minPs` here.

---

*[storage/replica_test.go, line 226 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7OKcV4j6dhsY_8FtDN-r1-226:-K7SZ9GErQ2kl4mfTxx8:2108631346) ([raw file](https://github.com/cockroachdb/cockroach/blob/c4b7420396de5550bd0374ce55faecd2cdfca624/storage/replica_test.go#L226)):*
personal preferences diverge (mine goes the other way, for example).

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-:-K7SZmgHSlmPcrliZGns:1922822350)*

**Comment by tamird** [comment]
Reviewed 6 of 6 files at r3.
Review status: all files reviewed at latest revision, 13 unresolved discussions, some commit checks failed.

---

*[roachpb/data.go, line 629 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7RqRIskqWISSP5CXZq:-K7SayiJjmFsk45fBLJk:1388305574) ([raw file](https://github.com/cockroachdb/cockroach/blob/84000e759288e4affd19ee84a602777e66979f95/roachpb/data.go#L629)):*
Sorry, typo. I meant that the ranges in this comment should be `[1, 2^31)`, not `[1, 2^31-1)` as currently written.

---

*[roachpb/data.go, line 693 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7RqjCf19PvDi4i0w_q:-K7SbC3L0ryp7sS34l00:-2047058171) ([raw file](https://github.com/cockroachdb/cockroach/blob/84000e759288e4affd19ee84a602777e66979f95/roachpb/data.go#L693)):*
you changed `userPriority` to `priority` in the comment, despite the variable in the code still being `userPriority`. old version seems better.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659)*

**Comment by spencerkimball** [comment]
Review status: 13 of 14 files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---

*[roachpb/api.proto, line 597 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7S55L9AA43uuMiMUwf:-K7SwWzK5_z2lr1jG3EZ:-1364079931) ([raw file](https://github.com/cockroachdb/cockroach/blob/84000e759288e4affd19ee84a602777e66979f95/roachpb/api.proto#L597)):*
OK, I'm going to have 0 mean the same thing as 1 and will make this nullable=false and drop default.

---

*[roachpb/data.go, line 612 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7SQE8O1t1uDrKCffB-:-K7SyRSqlo3q-UtMCi_B:-247507029) ([raw file](https://github.com/cockroachdb/cockroach/blob/84000e759288e4affd19ee84a602777e66979f95/roachpb/data.go#L612)):*
I don't understand your objection. userPriority = probWin / probLose is the same thing as "the likelihood of winning is userPriority * likelihood of losing. Maybe I'm missing something...?

---

*[roachpb/data.go, line 641 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7S6GcbCZum5zfWaIAL:-K7SzfFswX3NP4dco892:-1128321918) ([raw file](https://github.com/cockroachdb/cockroach/blob/84000e759288e4affd19ee84a602777e66979f95/roachpb/data.go#L641)):*
Updated to "pri"

---

*[roachpb/data.go, line 693 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7RqjCf19PvDi4i0w_q:-K7SzySjRlIKpsDUoCSy:466969942) ([raw file](https://github.com/cockroachdb/cockroach/blob/84000e759288e4affd19ee84a602777e66979f95/roachpb/data.go#L693)):*
Because the old comment was wrong. We're potentially upgrading the int32 txn priority based on rolling the dice again with the user priority as well as on the conflicting transaction's "upgradePriority".

---

*[roachpb/data_test.go, line 508 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7SWAEGJx1MdVaZCIdY:-K7T-9d85fGatDaOy3VF:-341030667) ([raw file](https://github.com/cockroachdb/cockroach/blob/84000e759288e4affd19ee84a602777e66979f95/roachpb/data_test.go#L508)):*
It's really handy when things aren't working properly. If this breaks again because we change things, I'll want to uncomment it.

---

*[storage/replica_test.go, line 226 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7OKcV4j6dhsY_8FtDN-r1-226:-K7T0FpLgx_E7LMXl-Kn:-1170764416) ([raw file](https://github.com/cockroachdb/cockroach/blob/c4b7420396de5550bd0374ce55faecd2cdfca624/storage/replica_test.go#L226)):*
Strange that go failed to define a standard here. Go's not very good with formatting line breaks (witness the presentation when breaking method arguments into multiple lines). We do have a lot of lines longer than 80 characters in our source code.  I was cleaning these up under the assumption people didn't care one way or another.  I'm happy to change it back, but we should consider adopting our own standard.

I should point out that this line is actually shorter than many of the lines in this file, so it seems inconsistent to break it arbitrarily.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659)*

**Comment by bdarnell** [comment]
Review status: 3 of 16 files reviewed at latest revision, 9 unresolved discussions.

---

*[client/db.go, line 177 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7T6NR4yOax9LH-ExlT:-K7T6NR4yOax9LH-ExlU:108836980) ([raw file](https://github.com/cockroachdb/cockroach/blob/9e2dfec169b57df7461f7299ce7f48071ec0ccef/client/db.go#L177)):*
Do we even need Sender-level priority anymore? 

---

*[client/db.go, line 479 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7T64DyDpEmV6jvrkdg:-K7T64DyDpEmV6jvrkdh:805730033) ([raw file](https://github.com/cockroachdb/cockroach/blob/9e2dfec169b57df7461f7299ce7f48071ec0ccef/client/db.go#L479)):*
0 and 1 should be OK, but using equality on floats still makes me nervous. 

Why can't we make 0 the special value for both fields instead of using 0 for one and 1 for the other?

---

*[storage/replica_test.go, line 226 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7OKcV4j6dhsY_8FtDN-r1-226:-K7T802201rlyLWyq_lH:845026837) ([raw file](https://github.com/cockroachdb/cockroach/blob/c4b7420396de5550bd0374ce55faecd2cdfca624/storage/replica_test.go#L226)):*
We have specified a 100-column limit in CONTRIBUTING.md.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659)*

**Comment by tamird** [comment]
Reviewed 13 of 13 files at r4.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[client/db.go, line 175 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7T_HqgxzqiOdwiT3YK:-K7T_HqgxzqiOdwiT3YL:-1318669080) ([raw file](https://github.com/cockroachdb/cockroach/blob/9e2dfec169b57df7461f7299ce7f48071ec0ccef/client/db.go#L175)):*
this comment should change back

---

*[roachpb/data.go, line 693 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7RqjCf19PvDi4i0w_q:-K7TYx8KjrQeaT1aVVkY:-470998475) ([raw file](https://github.com/cockroachdb/cockroach/blob/84000e759288e4affd19ee84a602777e66979f95/roachpb/data.go#L693)):*
> a new random priority using priority

what is the "priority" referred to here? (the second one)

---

*[roachpb/data_test.go, line 508 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7SWAEGJx1MdVaZCIdY:-K7TZEarsurj8SYhRCi7:1130633439) ([raw file](https://github.com/cockroachdb/cockroach/blob/84000e759288e4affd19ee84a602777e66979f95/roachpb/data_test.go#L508)):*
change it to `log.Info` and uncomment?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659)*

**Comment by tschottdorf** [comment]
LGTM

---

Reviewed 13 of 14 files at r2, 6 of 6 files at r3, 13 of 13 files at r4.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[storage/replica_test.go, line 226 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-K7OKcV4j6dhsY_8FtDN-r1-226:-K7T3jGM_i-IW5V1pBmF:618492871) ([raw file](https://github.com/cockroachdb/cockroach/blob/c4b7420396de5550bd0374ce55faecd2cdfca624/storage/replica_test.go#L226)):*
One reliable way to remove noisy edits related to this is not changing unrelated code. I have the impression that this has mostly eliminated back-and-forth between personal preferences of different folks.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3659#-:-K7UtIo2cWlJMazDKQjh:676834233)*
